plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -266,7 +266,14 @@ public IcebergMetadata(
     @Override
     public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
     {
-        IcebergTableName name = IcebergTableName.from(tableName.getTableName());
+        IcebergTableName name;
+        try {
+            name = IcebergTableName.from(tableName.getTableName());
+        }
+        catch (InvalidIcebergTableName e) {
+            log.debug(e, "Invalid table name when looking for a table");
```

Comment:
I don't think this behavior is useful. Let's change `IcebergTableName.from()` to return `Optional` and maybe rename it to `parse()`. The only reason we threw exceptions was to present them to the end user.

Reply:
I realized the exceptions were thoughtfully constructed, and so i didn;'t want to just discard those messages.
But then, presenting them directly to a user would be at odds with "table not found" expected behavior.
That's why i left them as debug output, in case there is any need to debug anything.

In the future, we can emit them as warnings. If it makes sense to you as well, i will add a TODO in the code.

Reply:
They were thoughtfully constructed because they were presented to end users. Since these are no longer displayed anywhere, I'd rather just remove the code.

Reply:
Actually, I'm not convinced that `NOT_FOUND` is necessarily the right behavior here. These are hidden tables, so there's nothing preventing us from saying that they do exist but are invalid. These would be similar to a Hive table that exists but has bad metadata, for which we would throw `HIVE_INVALID_METADATA`. Or how we throw `HIVE_UNSUPPORTED_FORMAT` for Delta Lake tables.